src/components/VHeader/VFilterButton.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <VButton
+    :variant="buttonVariant"
+    class="self-center"
+    :pressed="pressed"
+    aria-controls="filter-sidebar"
+    @click="toggleFilters"
+  >
+    <VIcon
+      v-show="showIcon"
+      :icon-path="filterIcon"
+      :class="{ 'me-2': iconNeedsPadding }"
+    />
+    <span class="filter-label" :class="buttonLabel.a11yProps">{{
+      buttonLabel.text
+    }}</span>
+  </VButton>
+</template>
+
+<script>
+import { computed, defineComponent, useContext } from '@nuxtjs/composition-api'
+import { isScreen } from '~/composables/use-media-query'
+import { useFilterSidebarVisibility } from '~/composables/use-filter-sidebar-visibility'
+
+import filterIcon from '~/assets/icons/filter.svg'
+import VButton from '~/components/VButton.vue'
+import VIcon from '~/components/VIcon/VIcon'
+
+const VFilterButton = defineComponent({
+  name: 'VFilterButton',
+  components: {
+    VIcon,
+    VButton,
+  },
+  props: {
+    /**
+     * Whether the page has been scrolled. If true on mobile, the button
+     * label is blank.
+     * This is currently not implemented, and is always false.
+     * TODO(@obulat): add use-scroll composable to VHeader
+     * @default false
+     */
+    isScrolled: {
+      type: Boolean,
+      default: false,
+    },
```

Comment:
Why does this component need this as a prop, rather than being able to use the `useScroll` composable itself to tell whether the page is scrolled?

Reply:
I guess my thinking at the time was that scroll position will be used by several header components, and could be set once in the header and passed down to children as props. I have to look if it's possible to set isScrolled as a global ref that could be used by all components...

Reply:
Ah I see, to avoid processing it multiple times? A global ref seems like it would be a good use case for that 😁 What a handy tool!